### PR TITLE
feature(wallet-mobile): New add wallet funnel revamp

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

**Comment by stackchain** [comment]
@SorinC6 can you please wire up the logic? My suggestion would be to create a package such as `@yoroi/transfer` -> `@yoroi/setup-wallet`

File: apps/wallet-mobile/src/AppNavigator.tsx [hunk 1]
```diff
@@ -19,11 +19,11 @@ import {ModalScreen} from './components/Modal/ModalScreen'
 import {AgreementChangedNavigator, InitializationNavigator} from './features/Initialization'
 import {LegalAgreement, useLegalAgreement} from './features/Initialization/common'
 import {useDeepLinkWatcher} from './features/Links/common/useDeepLinkWatcher'
+import {AddWalletNavigator} from './features/Wallet/AddWalletNavigator'
```
Comment (stackchain) [inline apps/wallet-mobile/src/AppNavigator.tsx]: SetupWalletNavigator

File: apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx [hunk 2]
```diff
@@ -7,11 +7,11 @@ import {SafeAreaView} from 'react-native-safe-area-context'
 import {KeyboardAvoidingView, ProgressStep, Spacer, TextInput} from '../components'
 import {ConfirmTx} from '../components/ConfirmTx'
 import {debugWalletInfo, features} from '../features'
+import {useSelectedWallet} from '../features/Wallet/common/Context'
```
Comment (stackchain) [inline apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx]: WalletSelection feature is part of WalletManager, it shouldn't be part of the wallet creation/restore.

Comment (stackchain) [reply apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx]: `features/SetupWallet/`

File: apps/wallet-mobile/src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -72,8 +71,6 @@ export class ErrorBoundary extends Component<Props, State> {
 
               <Spacer height={16} />
             </View>
-
```
Comment (stackchain) [inline apps/wallet-mobile/src/components/ErrorBoundary/ErrorBoundary.tsx]: this was removed by the vendor, when it shouldn't.

File: apps/wallet-mobile/src/features/Wallet/AddWalletNavigator.tsx [hunk 4]
```diff
@@ -1,61 +1,77 @@
...
-        ...defaultStackNavigationOptions,
+        ...navigationOptions,
         detachPreviousScreen: false /* https://github.com/react-navigation/react-navigation/issues/9883 */,
       }}
     >
       <Stack.Screen
-        name="initial-choose-create-restore"
-        component={WalletFreshInitScreen}
-        options={{headerShown: false}}
+        name="add-wallet-choose-setup-type"
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Wallet/AddWalletNavigator.tsx]: rename all routes from `add-wallet` to `setup-wallet-`

File: apps/wallet-mobile/src/features/Wallet/common/Context/SelectedWalletContext.tsx
Comment (stackchain) [inline apps/wallet-mobile/src/features/Wallet/common/Context/SelectedWalletContext.tsx:1]: SelectWallet is a different useCase from WalletSelection -> SetupWallet is a different feature, with useCases of create/restore wallet.

File: apps/wallet-mobile/src/features/Wallet/legacy/CheckNanoX/index.ts
Comment (stackchain) [inline apps/wallet-mobile/src/features/Wallet/legacy/CheckNanoX/index.ts:1]: delete this file.

File: apps/wallet-mobile/src/features/Wallet/legacy/ConnectNanoX/index.ts
Comment (stackchain) [inline apps/wallet-mobile/src/features/Wallet/legacy/ConnectNanoX/index.ts:1]: delete.

File: apps/wallet-mobile/src/features/Wallet/legacy/ImportReadOnlyWallet/index.ts
Comment (stackchain) [inline apps/wallet-mobile/src/features/Wallet/legacy/ImportReadOnlyWallet/index.ts:1]: delete

File: apps/wallet-mobile/src/features/Wallet/legacy/SaveNanoX/index.ts
Comment (stackchain) [inline apps/wallet-mobile/src/features/Wallet/legacy/SaveNanoX/index.ts:1]: delete it

**Comment by stackchain** [comment]
@banklesss I changed my mind, let's call it `setup-wallet` -> `features/SetupWallet` -> `SetupWalletNavigator`

**Comment by stackchain** [comment]
@banklesss let's merge this update after these tiny changes, and give it another round to load off the input -> password, input -> wallet name, to the state, using the same schema from transfer, so errors, display value, etc... will also be available x-platform.

**Comment by stackchain** [comment]
- [x] Rename navigator to `setup-wallet-<...>`
- [x] Rename package to `@yoroi/setup-wallet`
- [x] Rename feature to `src/features/SetupWallet`
- [x] Move back wallet selection stuff

File: apps/wallet-mobile/src/AppNavigator.tsx (hunk 1 above)
Comment (banklesss) [reply apps/wallet-mobile/src/AppNavigator.tsx]: done

File: apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx (hunk 2 above)
Comment (banklesss) [reply apps/wallet-mobile/src/Catalyst/ConfirmVotingTx.tsx]: done

File: apps/wallet-mobile/src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 3 above)
Comment (banklesss) [reply apps/wallet-mobile/src/components/ErrorBoundary/ErrorBoundary.tsx]: done

File: apps/wallet-mobile/src/features/Wallet/AddWalletNavigator.tsx (hunk 4 above)
Comment (banklesss) [reply apps/wallet-mobile/src/features/Wallet/AddWalletNavigator.tsx]: done

File: apps/wallet-mobile/src/features/Wallet/common/Context/SelectedWalletContext.tsx
Comment (banklesss) [reply apps/wallet-mobile/src/features/Wallet/common/Context/SelectedWalletContext.tsx:1]: done

File: apps/wallet-mobile/src/features/Wallet/legacy/CheckNanoX/index.ts
Comment (banklesss) [reply apps/wallet-mobile/src/features/Wallet/legacy/CheckNanoX/index.ts:1]: done

File: apps/wallet-mobile/src/features/Wallet/legacy/ConnectNanoX/index.ts
Comment (banklesss) [reply apps/wallet-mobile/src/features/Wallet/legacy/ConnectNanoX/index.ts:1]: done

File: apps/wallet-mobile/src/features/Wallet/legacy/ImportReadOnlyWallet/index.ts
Comment (banklesss) [reply apps/wallet-mobile/src/features/Wallet/legacy/ImportReadOnlyWallet/index.ts:1]: done

File: apps/wallet-mobile/src/features/Wallet/legacy/SaveNanoX/index.ts
Comment (banklesss) [reply apps/wallet-mobile/src/features/Wallet/legacy/SaveNanoX/index.ts:1]: done

**Pull request merged**